### PR TITLE
CEDS-4560 Add 'Amendment rejected' event to the Timeline Page

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: test/views/DeclarationDetailsViewSpec.scala [hunk 1]
```diff
@@ -557,6 +557,30 @@ class DeclarationDetailsViewSpec extends UnitViewSpec with GivenWhenThen with In
       }
     }
 
+    def createSubmissionWithRejectedAmendment(notificationSummaries: Seq[NotificationSummary]) = {
```
Comment (udinobi) [inline test/views/DeclarationDetailsViewSpec.scala:560]: Maybe you could avoid to create this new method. You already have `createSubmissionWith`, isn't it? You could you use it just modifying its signature...
```
def createSubmissionWith(notificationSummaries: Seq[NotificationSummary], requestType: RequestType = SubmissionRequest)
```

File: test/views/DeclarationDetailsViewSpec.scala [hunk 2]
```diff
@@ -557,6 +557,30 @@ class DeclarationDetailsViewSpec extends UnitViewSpec with GivenWhenThen with In
...
+        latestDecId = Some(uuid)
+      )
+    }
+
+    "display 'Fix and resubmit' button and 'Cancel' link when latest notification is rejected Amendment" in {
+      val notificationSummariesAmendmentRejected = List(dmsrejNotification, dmsdocNotification, dmsctlNotification, acceptedNotification)
+      val view = page(createSubmissionWithRejectedAmendment(notificationSummariesAmendmentRejected))(verifiedEmailRequest(), messages)
+      val button = view.getElementsByClass("govuk-button-group")
+      button.get(0).child(0).text() mustBe messages("declaration.details.fix.resubmit.button")
+      button.get(0).child(1).text() mustBe messages("declaration.details.cancel.amendment")
```
Comment (udinobi) [inline test/views/DeclarationDetailsViewSpec.scala:580]: I would also test in addition that `child(0)` is an actual button and that `child(1)` is just instead a link

File: test/views/DeclarationDetailsViewSpec.scala [hunk 3]
```diff
@@ -557,6 +557,30 @@ class DeclarationDetailsViewSpec extends UnitViewSpec with GivenWhenThen with In
...
+        notificationSummaries.headOption.map(_ => now),
+        Seq(action),
+        latestDecId = Some(uuid)
+      )
+    }
+
+    "display 'Fix and resubmit' button and 'Cancel' link when latest notification is rejected Amendment" in {
+      val notificationSummariesAmendmentRejected = List(dmsrejNotification, dmsdocNotification, dmsctlNotification, acceptedNotification)
+      val view = page(createSubmissionWithRejectedAmendment(notificationSummariesAmendmentRejected))(verifiedEmailRequest(), messages)
+      val button = view.getElementsByClass("govuk-button-group")
```
Comment (udinobi) [inline test/views/DeclarationDetailsViewSpec.scala:578]: Uhm... I would name it as `buttonGroup` or something like that.

File: test/views/helpers/TimelineEventsSpec.scala [hunk 4]
```diff
@@ -142,6 +152,31 @@ class TimelineEventsSpec extends UnitViewSpec with BeforeAndAfterEach with Injec
       timelineEvents(2).title mustBe messages(s"submission.enhancedStatus.$RECEIVED")
     }
 
+    "generate the expected sequence of TimelineEvent instances when the latest action is amendment is rejected" in {
+      val timelineEvents = createTimelineFromSubmission(amendmentRejectedLatest)
+
+      timelineEvents.size mustBe 4
+      timelineEvents(0).title mustBe messages("submission.enhancedStatus.timeline.title.amendment.rejected")
```
Comment (udinobi) [inline test/views/helpers/TimelineEventsSpec.scala:159]: You could also add a check for `timelineEvents(0).dateTime`

File: test/views/helpers/TimelineEventsSpec.scala [hunk 5]
```diff
@@ -142,6 +152,31 @@ class TimelineEventsSpec extends UnitViewSpec with BeforeAndAfterEach with Injec
...
+
+    "generate the expected sequence of TimelineEvent instances when the amendment is rejected" in {
+      val timelineEvents = createTimelineFromSubmission(amendmentRejected)
+
+      timelineEvents.size mustBe 4
+
+      timelineEvents(0).title mustBe messages(s"submission.enhancedStatus.$ERRORS")
+      timelineEvents(1).title mustBe messages(s"submission.enhancedStatus.$GOODS_ARRIVED_MESSAGE")
+      timelineEvents(2).title mustBe messages(s"submission.enhancedStatus.$RECEIVED")
+      timelineEvents(3).title mustBe messages("submission.enhancedStatus.timeline.title.amendment.rejected")
```
Comment (udinobi) [inline test/views/helpers/TimelineEventsSpec.scala:175]: Same here

File: test/views/DeclarationDetailsViewSpec.scala [hunk 6]
```diff
@@ -557,6 +557,20 @@ class DeclarationDetailsViewSpec extends UnitViewSpec with GivenWhenThen with In
       }
     }
 
+    "display 'Fix and resubmit' button and 'Cancel' link when latest notification is rejected Amendment" in {
+      val notificationSummariesAmendmentRejected = List(dmsrejNotification, dmsdocNotification, dmsctlNotification, acceptedNotification)
+      val view = page(createSubmissionWith(notificationSummariesAmendmentRejected, AmendmentRequest))(verifiedEmailRequest(), messages)
+      val buttonGroup = view.getElementsByClass("govuk-button-group")
+      val button = buttonGroup.get(0).child(0).text()
+      val link = buttonGroup.get(0).child(1).text()
+
```
Comment (udinobi) [inline test/views/DeclarationDetailsViewSpec.scala:566]: Ouch... this is wrong. `button` and `link` have `String` type, they are not Html `Element`. Code here should be:
```
      val button = buttonGroup.get(0).child(0)
      val link = buttonGroup.get(0).child(1)

      button.text() mustBe messages("declaration.details.fix.resubmit.button")
      button.hasClass("govuk-button")

      link.text() mustBe messages("declaration.details.cancel.amendment")
      link.hasClass("gov-link")
```

**Review by magicaljellybeans — APPROVED** [review]

**Pull request merged**